python/cudf/cudf/utils/utils.py

```diff
@@ -277,25 +276,25 @@ def _set_rmm_config(
         If ``False`` (default), disable pool mode.
     initial_pool_size : int, optional
         If ``use_pool_allocator=True``, sets initial pool size.
-        If ``None``, us
-es 1/2 of total GPU memory.
+        If ``None``, uses 1/2 of total GPU memory.
     enable_logging : bool, optional
         Enable logging (default ``False``).
         Enabling this option will introduce performance overhead.
     """
-    rmm.finalize()
-    rmm_config.use_managed_memory = use_managed_memory
     if use_pool_allocator:
-        rmm_config.use_pool_allocator = use_pool_allocator
         if initial_pool_size is None:
             initial_pool_size = 0  # 0 means 1/2 GPU memory
         elif initial_pool_size == 0:
             initial_pool_size = 1  # Since "0" is semantic value, use 1 byte
```

Comment:
I believe some of this logic was introduced in rapidsai/rmm#171 which makes it tough to initialize pool mode with 1/2 gpu memory from cudf. `initial_pool_size=None` sets it to 0 here. rapidsai/rmm#171 infers `initial_pool_size` as 0 and sets it to 1, hence making it tough to initialize with the default 1/2 Gpu memory case